solr/core/src/java/org/apache/solr/search/facet/FacetRangeProcessor.java

```diff
@@ -278,6 +329,24 @@ private void createRangeList() throws IOException {
     }
   }
 
+  @SuppressWarnings({"unchecked"})
+  private void sortRangeList() {
+    rangeList.sort((r1, r2) -> {
+      if (r1.low == null) {
+        return r2.low == null ? 0 : -1;
+      }
+      if (r2.low == null) {
+        return 1;
+      }
+      int res = r1.low.compareTo(r2.low);
+      if (res == 0) {
+        // tie break goes to if either include lower as that is technically a lower low value
+        return r1.includeLower == r2.includeLower ? 0 : (r1.includeLower ? -1 : 1);
```

Comment:
Right, because the thing that doesn't include lower actually only includes data points higher. It's kind of an annoying tie break. This has also moved in the latest changes.